src/everest/gui/main_window.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import ssl
+
+from PyQt6.QtCore import pyqtSignal as Signal
+from PyQt6.QtWidgets import (
+    QFrame,
+    QHBoxLayout,
+    QMainWindow,
+)
+
+from ert.gui.ertnotifier import ErtNotifier
+from ert.gui.simulation.run_dialog import RunDialog
+from ert.plugins import ErtPluginManager
+from everest.config import EverestConfig, ServerConfig
+from everest.detached import wait_for_server
+from everest.gui.everest_client import EverestClient
+
+
+class EverestMainWindow(QMainWindow):
+    close_signal = Signal()
+
+    def __init__(
+        self,
+        config_file: str,
+    ):
+        QMainWindow.__init__(self)
+        self.config_file = config_file
+
+        self.setWindowTitle(f"Everest - {config_file}")
+        self.plugin_manager = ErtPluginManager()
+        self.central_widget = QFrame(self)
+        self.central_layout = QHBoxLayout(self.central_widget)
+        self.central_layout.setContentsMargins(0, 0, 0, 0)
+        self.central_layout.setSpacing(0)
+        self.central_widget.setLayout(self.central_layout)
+
+        self._run_dialog: RunDialog | None = None
+
+        self.central_widget.setMinimumWidth(1500)
+        self.central_widget.setMinimumHeight(800)
+        self.setCentralWidget(self.central_widget)
+
+    def post_init(self) -> None:
```

Comment:
Suggest a renamve to for example `run`, `post_init` sounds like it should not have as many side effects.